src/jit/emitxarch.cpp

```diff
@@ -3573,10 +3573,24 @@ void emitter::emitInsRMW(instruction ins, emitAttr attr, GenTreeStoreInd* storeI
     if (src->isContainedIntOrIImmed())
     {
         GenTreeIntConCommon* intConst = src->AsIntConCommon();
-        id                            = emitNewInstrAmdCns(attr, offset, (int)intConst->IconValue());
+        int                  iconVal  = (int)intConst->IconValue();
+        switch (ins)
+        {
+            case INS_rcl_N:
+            case INS_rcr_N:
+            case INS_rol_N:
+            case INS_ror_N:
+            case INS_shl_N:
+            case INS_shr_N:
+            case INS_sar_N:
+                iconVal &= 0x7F;
```

Comment:
I would think we should either support all values (just sign-extend the icon node) or we should mask in a way that matches what the hardware is doing:
> The count is masked to 5 bits (or 6 bits if in 64-bit mode and REX.W = 1).

Just sign-extending the icon node seems like the overall safer option.

Reply:
The masking is done like this in many places in this file. If the constants should be sign-extended instead I would prefer to do it consistently in the file.
For example:
https://github.com/dotnet/coreclr/blob/master/src/jit/emitxarch.cpp#L3802-L3814

Reply:
Right, I think it's best to be consistent.